src/contracts/pods/EigenPod.sol

```diff
@@ -488,274 +451,182 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-        );
-
-        // 2. Balance updates should only be performed on "ACTIVE" validators
-        require(
-            validatorInfo.status == VALIDATOR_STATUS.ACTIVE, 
-            "EigenPod.verifyBalanceUpdate: Validator not active"
-        );
+    function _verifyCheckpointProof(
+        ValidatorInfo memory validatorInfo,
+        uint64 checkpointTimestamp,
+        bytes32 balanceContainerRoot,
+        BeaconChainProofs.BalanceProof calldata proof
+    ) internal returns (int128 balanceDeltaGwei, uint64 exitedBalanceGwei) {
+        uint40 validatorIndex = uint40(validatorInfo.validatorIndex);
+        
+        // Verify validator balance against `balanceContainerRoot`
+        uint64 prevBalanceGwei = validatorInfo.restakedBalanceGwei;
+        uint64 newBalanceGwei = BeaconChainProofs.verifyValidatorBalance({
+            balanceContainerRoot: balanceContainerRoot,
+            validatorIndex: validatorIndex,
+            proof: proof
+        });
+        
+        // Calculate change in the validator's balance since the last proof
+        if (newBalanceGwei != prevBalanceGwei) {
+            balanceDeltaGwei = _calcBalanceDelta({
+                newAmountGwei: newBalanceGwei,
+                previousAmountGwei: prevBalanceGwei
+            });
 
-        // 3. Balance updates should only be made before a validator is fully withdrawn. 
-        // -- A withdrawable validator may not have withdrawn yet, so we require their balance is nonzero
-        // -- A fully withdrawn validator should withdraw via verifyAndProcessWithdrawals
-        if (validatorFields.getWithdrawableEpoch() <= _timestampToEpoch(oracleTimestamp)) {
-            require(
-                validatorEffectiveBalanceGwei > 0,
-                "EigenPod.verifyBalanceUpdate: validator is withdrawable but has not withdrawn"
-            );
+            emit ValidatorBalanceUpdated(validatorIndex, checkpointTimestamp, newBalanceGwei);
         }
 
-        // Verify passed-in validatorFields against verified beaconStateRoot:
-        BeaconChainProofs.verifyValidatorFields({
-            beaconStateRoot: beaconStateRoot,
-            validatorFields: validatorFields,
-            validatorFieldsProof: validatorFieldsProof,
-            validatorIndex: validatorIndex
-        });
+        validatorInfo.restakedBalanceGwei = newBalanceGwei;
+        validatorInfo.lastCheckpointedAt = checkpointTimestamp;
 
-        // Done with proofs! Now update the validator's balance and send to the EigenPodManager if needed
+        // If the validator's new balance is 0, mark them withdrawn
+        if (newBalanceGwei == 0) {
+            activeValidatorCount--;
+            validatorInfo.status = VALIDATOR_STATUS.WITHDRAWN;
+            // If we reach this point, `balanceDeltaGwei` should always be negative,
+            // so this should be a safe conversion
+            exitedBalanceGwei = uint64(uint128(-balanceDeltaGwei));
```

Comment:
noting that this may slashed and not actually in the pod

Reply:
i have a giant disclaimer comment above the `checkpointBalanceExitedGwei` state var - have a look through that and lmk if you think it needs more.

Reply:
lgtm